### PR TITLE
Minimal spanning network

### PR DESCRIPTION
**Pull request by dwhswenson**
Trying to fill in the stub for `minimal_spanning_graph`.

I think this is working now (has a basic test that covers code and checks the things I could think of as reasonable).

This includes #137, and so should only be merged after that.

I'd welcome any additions to the tests.

### CODE REVIEW

**Comment by dwhswenson** [comment]
I think this is ready for review. I still feel like it is somewhat under-tested, so I'd appreciate suggestions on that.

Question: What should happen if a given node is unreachable (no edge is created that connects it to the network)? Currently, I think that would pass silently (you get an unconnected node in the output network). Should that error instead? Raise a warning?

File: openfe/setup/ligand_network_planning.py
Comment (richardjgowers) [inline openfe/setup/ligand_network_planning.py:99]: I can't remember if we ever decided if mappers are reversible/symmetric, this is obviously assuming A->B === B->A.  @IAlibay are mappings **always** symmetric?

File: openfe/setup/ligand_network_planning.py
Comment (richardjgowers) [inline openfe/setup/ligand_network_planning.py:95]: please non non-ascii in the source

File: openfe/setup/ligand_network_planning.py
Comment (richardjgowers) [inline openfe/setup/ligand_network_planning.py:98]: style quibble, keep this as a generator expression and evaluate it in the next step?

**Review by richardjgowers — CHANGES_REQUESTED** [review]
LGTM, I think maybe once the lomap scorers are in we can have a "real" example to regression test onto too, and it might be worth holding off till then

File: openfe/setup/ligand_network_planning.py
Comment (IAlibay) [reply openfe/setup/ligand_network_planning.py:99]: As far as I'm aware they should be, otherwise there's a gap in the free energy between the forward and reverse transformation.

**Comment by richardjgowers** [comment]
@dwhswenson and re: unreachable node, this should be an error for this function, but not necessarily in the Network object.  I imagine nx gives an error if it can't construct the MST and we should catch & nicely format that

**Review by IAlibay — APPROVED** [review]
lgtm!

@richardjgowers is that LGTM meant to be blocking?

**Comment by dwhswenson** [comment]
One remaining possible blocker is to add the test for unreachable nodes (which I can do this afternoon).

File: openfe/setup/ligand_network_planning.py
Comment (richardjgowers) [inline openfe/setup/ligand_network_planning.py:109]: does nx silently provide a non spanning network if there were no edges for a particular node?

Comment (dwhswenson) [reply openfe/setup/ligand_network_planning.py:109]: Yes. Test added in https://github.com/OpenFreeEnergy/openfe/pull/139/commits/a4a30e3c5dc7730033e17066674ef89f256b544f shows that it would happen -- we catch the error I added, not something from nx.

Maybe this is a difference between `nx.minimum_spanning_edges` vs. some of the other related nx functions?

**Review by richardjgowers — APPROVED** [review]

**Pull request merged**
